external/java/src/main/java/ch/epfl/dedis/skipchain/SkipchainRPC.java

```diff
@@ -86,6 +86,12 @@ public SkipBlock getSkipblock(SkipblockId id) throws CothorityCommunicationExcep
             SkipchainProto.SkipBlock sb = SkipchainProto.SkipBlock.parseFrom(msg);
             SkipBlock ret = new SkipBlock(sb);
 
+            if (!Arrays.equals(id.getId(), ret.getHash())) {
+                // we don't want a malicious peer to send a block with tempered
+                // data like a different roster with different public keys
+                throw new CothorityCommunicationException("invalid block integrity");
+            }
```

Comment:
the function comment says the block is not verified, is that still the case?

Reply:
I'll change the comment. The integrity is verified but the caller must verify if it belongs to the chain.